codalab/server/bundle_manager.py

```diff
@@ -436,11 +430,21 @@ def get_sort_key(worker):
             else:
                 deps = set(worker['dependencies'])
                 num_available_deps = len(needed_deps & deps)
-            worker_id = worker['worker_id']
 
-            # if the bundle doesn't request GPUs (only request CPUs), prioritize workers that don't have GPUs
-            gpu_priority = bundle_resources.gpus or not has_gpu[worker_id]
-            return (gpu_priority, num_available_deps, worker['cpus'], random.random())
+            # If the bundle requested for GPUs, prioritize only those workers that have GPUs.
+            # If the bundle didn't request for GPUs (a CPU only job), prioritize all workers based criteria as follows:
+            # 1. number of running jobs.
+            # 2. number of available dependencies.
```

Comment:
Number of available dependencies should be prioritized as high as possible (ideally the sum of the sizes of the dependencies) because we want to leverage the local bundle cache as much as possible, of course subject to being able to run a job at all.